### PR TITLE
docs tab for Collection Detail

### PR DESCRIPTION
**Pull request by ShaiahWren**
Draft pr for the Documentation tab on Collection Detail. 

Screenshot 2023-11-13 at 1 40 39 PM

### CODE REVIEW

File: frontend/hub/collections/CollectionDetails.tsx [hunk 1]
```diff
@@ -368,16 +369,24 @@ function CollectionDocumentationTab(props: { collection?: CollectionVersionSearc
...
-  const { data } = useGet<CollectionDocs>(
-    hubAPI`/_ui/v1/repo/published/${collection?.collection_version?.name || ''}/${
-      collection?.collection_version?.name || ''
-    }/?include_related=my_permissions`
+  const { data } = useGet<CollectionVersionListResponse>(
+    pulpAPI`/content/ansible/collection_versions/?namespace=${
+      collection?.collection_version?.namespace || ''
+    }&name=${collection?.collection_version?.name || ''}&version=${
+      collection?.collection_version?.version || ''
+    }&offset=0&limit=10`
```
Comment (himdel) [inline frontend/hub/collections/CollectionDetails.tsx:377]: might want to use `limit=1` if we're only looking at `data.results[0]`

File: frontend/hub/collections/CollectionDetails.tsx [hunk 2]
```diff
@@ -368,16 +369,24 @@ function CollectionDocumentationTab(props: { collection?: CollectionVersionSearc
 
   const [content, setContent] = useState<IContents>();
 
-  const { data } = useGet<CollectionDocs>(
-    hubAPI`/_ui/v1/repo/published/${collection?.collection_version?.name || ''}/${
-      collection?.collection_version?.name || ''
-    }/?include_related=my_permissions`
+  const { data } = useGet<CollectionVersionListResponse>(
```
Comment (nixocio) [inline frontend/hub/collections/CollectionDetails.tsx:372]: We could add error, and isLoading here.

Comment (MilanPospisil) [reply frontend/hub/collections/CollectionDetails.tsx:372]: You can look here https://issues.redhat.com/browse/AAP-15074. This is the way we want to handle errors using useGet.

File: frontend/hub/collections/CollectionDetails.tsx [hunk 3]
```diff
@@ -368,16 +369,24 @@ function CollectionDocumentationTab(props: { collection?: CollectionVersionSearc
...
+    pulpAPI`/content/ansible/collection_versions/?namespace=${
+      collection?.collection_version?.namespace || ''
+    }&name=${collection?.collection_version?.name || ''}&version=${
+      collection?.collection_version?.version || ''
+    }&offset=0&limit=1`
   );
 
+  const readme = useMemo(() => {
```
Comment (MilanPospisil) [inline frontend/hub/collections/CollectionDetails.tsx:380]: @himdel Do you think this useMemo is necessary? The data comes from useGet, which memoizes the data anyway, so direct reference to the readme blob should be always the same.

Comment (himdel) [reply frontend/hub/collections/CollectionDetails.tsx:380]: oh, good catch, but moot point, `readme` itself is unused :) we can remove the whole block

**Comment by MilanPospisil** [comment]
This has to be rebased, because of changes in collection tabs. Maybe it will be faster to create new PR and copy changes from this PR rather than rebase it.

**Comment by ShaiahWren** [comment]
Replacing with new pr for docs tab.

**Pull request closed without merging**